unstructured/ingest/connector/jira.py

```diff
@@ -0,0 +1,462 @@
+    return f"""
+    {issue["summary"]}
+    {r_sep}
+    {issue["description"]}
+    {r_sep}
+    {c_sep.join([atch["self"] for atch in issue["attachment"]])}
+    {r_sep}
+    """
+
+
+def _get_comments_for_issue(issue, c_sep=DEFAULT_C_SEP, r_sep=DEFAULT_R_SEP):
+    return c_sep.join(
+        [_get_fields_for_comment(comment) for comment in issue["comment"]["comments"]],
+    )
+
+
+def _get_fields_for_comment(comment, c_sep=DEFAULT_C_SEP, r_sep=DEFAULT_R_SEP):
+    return f"{comment['author']['displayName']}{c_sep}{comment['body']}{r_sep}"
+
+
+def scroll_wrapper(func, results_key="results"):
+    def wrapper(*args, **kwargs):
+        """Wraps a function to obtain scroll functionality.
+        Function needs to be able to accept 'start' and 'limit' arguments."""
+        if "number_of_items_to_fetch" in kwargs:
+            number_of_items_to_fetch = kwargs["number_of_items_to_fetch"]
+            del kwargs["number_of_items_to_fetch"]
+        else:
+            number_of_items_to_fetch = 100
+
+        kwargs["limit"] = min(100, number_of_items_to_fetch)
+        kwargs["start"] = 0 if "start" not in kwargs else kwargs["start"]
+
+        all_results = []
+        num_iterations = math.ceil(number_of_items_to_fetch / kwargs["limit"])
+
+        for _ in range(num_iterations):
+            response = func(*args, **kwargs)
+            if type(response) is list:
+                all_results += func(*args, **kwargs)
+            elif type(response) is dict:
+                if results_key not in response:
+                    raise KeyError(
+                        "Response object has no known keys to \
+                                   access the results, such as 'results' or 'values'.",
+                    )
+                all_results += func(*args, **kwargs)[results_key]
+            kwargs["start"] += kwargs["limit"]
+
+        return all_results[:number_of_items_to_fetch]
+
+    return wrapper
+
+
+@dataclass
+class JiraIngestDoc(IngestDocSessionHandleMixin, IngestDocCleanupMixin, BaseIngestDoc):
+    """Class encapsulating fetching a doc and writing processed results (but not
+    doing the processing).
+
+    Current implementation creates a Jira connection object
+    to fetch each doc, rather than creating a it for each thread.
+    """
+
+    config: SimpleJiraConfig
+    file_meta: JiraFileMeta
+    registry_name: str = "jira"
+
+    @cached_property
+    def source_url(self):
+        return f"{self.config.url}/browse/{self.file_meta.issue_key}"
+
+    @cached_property
+    def record_locator(self):  # Values must be JSON-serializable
+        """A dictionary with any data necessary to uniquely identify the document on
+        the source system."""
+        return {"issue_key": self.file_meta.issue_key}
+
+    @cached_property
+    @SourceConnectionError.wrap
+    @requires_dependencies(dependencies=["atlassian"], extras="jira")
+    def get_metadata_fields(self):
```

Comment:
nit: just call this `metadata_fields` since it is a property getter

Reply:
Addressed with https://github.com/Unstructured-IO/unstructured/pull/1238/commits/9bf2990aa42006204a5285502ab3d3f69d477794